gnoi/cert/manager.go

```diff
@@ -52,11 +59,17 @@ type Manager struct {
 }
 
 // NewManager returns a Manager.
-func NewManager(privateKey crypto.PrivateKey) *Manager {
+func NewManager(settings *Settings) *Manager {
+	certInfo := map[string]*Info{}
+	var privateKey crypto.PrivateKey
```

Comment:
Private key is always required. If tls.certificate is empty, then generate a key.